README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
 https://github.com/zulip/zulip/issues/32 for a plan for how to
 dramatically reduce this requirement).
 
-Start by cloning this repository: `git clone https://github.com/zulip/zulip.git`
+Contents:
+
+* [First-time contributors start here](#first-time-contributors-start-here)
+* [Using Vagrant](#using-vagrant)
+* [Using provision.py without Vagrant](#using-provisionpy-without-vagrant)
+* [By hand](#by-hand)
+* [Using Docker (experimental)](#using-docker-experimental)
```

Comment:
https://github.com/christi3k/zulip/blob/issue-747-windows-docs/README.dev.md#using-the-development-environment and https://github.com/christi3k/zulip/blob/issue-747-windows-docs/README.dev.md#running-the-test-suite should probably be in the contents?

(Also worth noting https://github.com/christi3k/zulip/blob/issue-747-windows-docs/README.dev.md#possible-testing-issues -- the first item is low-value, but the second should probably get moved into your troubleshooting section)